src/simulator.ts

```diff
@@ -88,13 +89,51 @@ export class Simulator {
 
   public async cancel() {
     this.cancelToken = true;
+    await this.delay(1000);
```

Comment:
We should already removed the delay in last pull request?

Reply:
Maybe that commit is not pushed. Solved this issue again.